TESTS/netsocket/udp/udpsocket_echotest.cpp

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2018, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mbed.h"
+#include MBED_CONF_APP_HEADER_FILE
+#include "UDPSocket.h"
+#include "greentea-client/test_env.h"
+#include "unity/unity.h"
+#include "utest.h"
+#include "udp_tests.h"
+
+#define SIGNAL_SIGIO 0x1
+#define SIGIO_TIMEOUT 5000 //[ms]
+#define WAIT2RECV_TIMEOUT 1000 //[ms]
+#define RETRIES 2
+
+static volatile bool running = true;
+
+UDPSocket sock;
+Semaphore tx_sem(0, 1);
+
+namespace
+{
+    static const int BUFF_SIZE = 1200;
+    char rx_buffer[BUFF_SIZE] = {0};
+    char tx_buffer[BUFF_SIZE] = {0};
+}
+
+static void _sigio_handler(osThreadId id) {
+    osSignalSet(id, SIGNAL_SIGIO);
+}
+
+void test_udpsocket_echotest()
+{
+    SocketAddress udp_addr;
+    get_interface()->gethostbyname(MBED_CONF_APP_ECHO_SERVER_ADDR, &udp_addr);
+    udp_addr.set_port(MBED_CONF_APP_ECHO_SERVER_PORT);
+
+    UDPSocket sock;
+    TEST_ASSERT_EQUAL(NSAPI_ERROR_OK, sock.open(get_interface()));
+
+    static const int PKTS = 15;
+    int pkt_sizes[PKTS] = {1,2,3,4,5,6,7,8,9,10,100,10,100,1200,100};
+
+    int recvd;
+    int sent;
+    int s_idx = 0;
+    for (int pkt_s = pkt_sizes[s_idx]; s_idx < PKTS; pkt_s = ++s_idx) {
+        pkt_s = pkt_sizes[s_idx];
+
+        fill_tx_buffer_ascii(tx_buffer, BUFF_SIZE);
```

Comment:
To be sure, clear the rx_buffer, so if we re-send same sized packet with same data, it will be shown.
`memset(rx_buffer, 0, BUFF_SIZE);